drivers/mtd/gd25.c

```diff
@@ -53,6 +53,26 @@
 #  define CONFIG_GD25_SPIFREQUENCY 20000000
 #endif
 
+#ifdef CONFIG_SPI_DELAY_CONTROL
+#  define CONFIG_GD25_USE_DELAY_CONTROL
+#endif
+
+#ifdef CONFIG_GD25_USE_DELAY_CONTROL
```

Comment:
This logic is not needed.  They will be defined if `CONFIG_SPI_DELAY_CONTROL` is defined. If someone manually played with the config file (which is not supported and breaks other things) you would still fail at build time indicating something is wrong.

Reply:
I was trying to make it bomb-proof but you're right. And with no need for that I don' need the `#  define CONFIG_GD25_USE_DELAY_CONTROL` either and can just use the `CONFIG_SPI_DELAY_CONTROL` directly